app/models/report.rb

```diff
@@ -14,4 +22,29 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def save_with_mentioning_reports
+    save
+    add_records_of_mentioning_reports
+  end
+
+  def update_with_mentioning_reports(report_params)
+    update(report_params)
+    add_records_of_mentioning_reports
+  end
+
+  def add_records_of_mentioning_reports
+    mentioning_reports.destroy_all if mentioning_reports.any?
+
+    scan_mentioning_report_ids.each do |id|
+      mentioning_report = Report.find(id)
+      mentioning_reports << mentioning_report unless mentioning_reports.include?(mentioning_report)
+    end
+  end
+
+  def scan_mentioning_report_ids
+    ids = content.scan(REPORT_ID_REGEXP).flatten
+
+    ids.map(&:to_i)
```

Comment:
```suggestion
    content.scan(REPORT_ID_REGEXP).flatten.map(&:to_i)
```

メソッドチェーンを使ってもいいかもしれません（好みの問題）

Reply:
メソッドチェーンを利用した方がすっきりしますね。
ご指摘の通り修正しました。